### PR TITLE
[984] Add Overview Page

### PR DESCRIPTION
**Pull request by keithchong**
Fixes https://github.com/eclipse/codewind/issues/984

Signed-off-by: Keith Chong <kchong@ca.ibm.com>

### CODE REVIEW

**Comment by keithchong** [comment]
@jopit , please review.  See screenshot from 984.  Note there is absolutely no code associated with the layouts or contraints.

File: dev/src/main/java/org/eclipse/codewind/intellij/ui/form/AppOverviewFrame.java [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+        statusField = new JTextField(); // Status
+
+        // Project Settings Pane
+        applicationContextRootField = new JTextField(); // Application Context Root
+        internalApplicationPortField = new JTextField(); // Internal Application Port
+        internalDebugPortField = new JTextField(); // Internal Debug Port
+
+        editProjectSettingsButton = new JButton();
+        editProjectSettingsButton.setText(message("AppOverviewEditorEditProjectSettings"));
+        editProjectSettingsButton.setToolTipText("Edit Codewind settings file"); // Todo: Externalize
```
Comment (jopit) [inline dev/src/main/java/org/eclipse/codewind/intellij/ui/form/AppOverviewFrame.java]: We should do the todo and externalize this now.

File: dev/src/main/java/org/eclipse/codewind/intellij/ui/form/AppOverviewFrame.java [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+                File cwSettings = new File(app.fullLocalPath + File.separator + CoreConstants.SETTINGS_FILE);
+                VirtualFile virtualFile = VfsUtil.findFileByIoFile(cwSettings, true);
+                if (virtualFile != null && virtualFile.exists()) {
+                    FileEditorManagerEx.getInstanceEx(proj).openFile(virtualFile, true);
+                }
+            }
+        });
+        moreInformationButton = new JButton();
+        moreInformationButton.setText(message("AppOverviewEditorProjectSettingsInfo"));
+        moreInformationButton.setToolTipText("Launch Codewind help");  // Todo: Externalize
```
Comment (jopit) [inline dev/src/main/java/org/eclipse/codewind/intellij/ui/form/AppOverviewFrame.java]: Again, we should externalize this now.

File: dev/src/main/java/org/eclipse/codewind/intellij/ui/form/AppOverviewFrame.java [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+        });
+
+        // Build Pane
+        autoBuildField = new JTextField(); // Auto Build
+        autoMetricsInjectionField = new JTextField(); // Auto Metrics Injection
+        lastBuildField = new JFormattedTextField(); // Last Build
+        lastImageBuildField = new JTextField(); // Last Image Build
+
+        // Bottom footer
+        preferencesLink = new JLabel("Control opening of overview page on project create and add");
```
Comment (jopit) [inline dev/src/main/java/org/eclipse/codewind/intellij/ui/form/AppOverviewFrame.java]: This should also be externalized

File: dev/src/main/java/org/eclipse/codewind/intellij/ui/tasks/OpenAppOverviewTask.java [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    public void run(@NotNull ProgressIndicator indicator) {
+        final ToolWindowManager toolWindowManager = ToolWindowManager.getInstance(getProject());
+        CoreUtil.invokeLater(new Runnable() {
+            @Override
+            public void run() {
+                ToolWindow overviewToolWindow = toolWindowManager.getToolWindow("Codewind Project Overview");
+                Content content = null;
+                ContentManager contentManager = null;
+                if (overviewToolWindow == null) {
+                    overviewToolWindow = toolWindowManager.registerToolWindow("Codewind Project Overview", true, ToolWindowAnchor.BOTTOM);
```
Comment (jopit) [inline dev/src/main/java/org/eclipse/codewind/intellij/ui/tasks/OpenAppOverviewTask.java]: Make "Codewind Project Overview" a constant

File: dev/src/main/java/org/eclipse/codewind/intellij/ui/form/AppOverviewFrame.java [hunk 5]
```diff
@@ -0,0 +1,322 @@
...
+    }
+
+    private void createUIComponents() {
+        final CodewindApplication app = this.application;
+        final Project proj = this.project;
+        overviewPanel = new JPanel();
+        overviewPanel.setBackground(JBColor.background());
+        // Top section showing app name
+        appLabel = new JLabel(this.application.name);
+        appLabel.setIcon(IconCache.getCachedIcon(IconCache.ICONS_CODEWIND_PNG));
```
[the remark this answers is not in the document]
Comment (keithchong) [reply dev/src/main/java/org/eclipse/codewind/intellij/ui/form/AppOverviewFrame.java]: That works.  Updated.

File: dev/src/main/java/org/eclipse/codewind/intellij/ui/form/AppOverviewFrame.java (hunk 1 above)
Comment (keithchong) [reply dev/src/main/java/org/eclipse/codewind/intellij/ui/form/AppOverviewFrame.java]: Done

File: dev/src/main/java/org/eclipse/codewind/intellij/ui/form/AppOverviewFrame.java (hunk 2 above)
Comment (keithchong) [reply dev/src/main/java/org/eclipse/codewind/intellij/ui/form/AppOverviewFrame.java]: Done

File: dev/src/main/java/org/eclipse/codewind/intellij/ui/form/AppOverviewFrame.java (hunk 3 above)
Comment (keithchong) [reply dev/src/main/java/org/eclipse/codewind/intellij/ui/form/AppOverviewFrame.java]: Done

File: dev/src/main/java/org/eclipse/codewind/intellij/ui/tasks/OpenAppOverviewTask.java (hunk 4 above)
Comment (keithchong) [reply dev/src/main/java/org/eclipse/codewind/intellij/ui/tasks/OpenAppOverviewTask.java]: Done

**Comment by keithchong** [comment]
@jopit, feedback changes have been made.

**Pull request merged**